src/lib.rs

```diff
@@ -212,10 +212,9 @@ fn filename_ext_to_xml(filename: String) -> Option<String>
         None
     } else {
         let path = Path::new(&filename);
-        let file_stem = path.file_stem();
-        let mut path2 = PathBuf::from(file_stem.unwrap().to_os_string());
-        path2.set_extension("xml");
-        Some(path2.into_os_string().into_string().unwrap())
+        let mut file_stem = path.file_stem().unwrap().to_os_string();
```

Comment:
Ik ga er vanuit dat die filename ook (een deel van) de folderstructuur kan bevatten. Want als dat sowieso de `basename` is, heb je de `stem` niet nodig en kan je gewoon `set_extension()` of `with_extension()` oproepen.